artifact.nix

```diff
@@ -29,10 +30,11 @@ let
   # Figure out version of bindist
   version =
     let
-      helper = stdenv.mkDerivation {
+      hostPkgs = import pkgs.path { config = {}; overlays = []; };
+      helper = hostPkgs.stdenv.mkDerivation {
         name = "bindist-version";
         src = bindistTarballs.${stdenv.targetPlatform.system};
-        nativeBuildInputs = [ gcc perl ];
+        nativeBuildInputs = [ hostPkgs.perl ];
```

Comment:
I thought putting things in `nativeBuildInputs` automatically made them be taken from the host packages? Or is that only if you use `callPackage`? Would it be better to stick to that idiom?

Reply:
I would have thought so to... alternatively, can't we just use `with pkgs.buildPackages; [ gcc perl ]`?

Reply:
`nativeBuildInputs` only works when you have a cross-compile

but this isnt an linux->darwin cross, its a plain darwin->darwin build

and then your doing IFD, and now you need a darwin build machine to unpack a tar file and read 1 plaintext file

so we need to bypass the normal framework, and allow the arch doing the eval to leak in

Reply:
a better solution for `bindist-version` would be to just put the answer into the `bindistTarballs` table, and not compute it at runtime at all

Reply:
So basically nix needs a cross-compilation *quadruple*, adding the eval arch, which might be different from the build arch. Yikes.

Although then I'm confused as to why `hostPackages` works - if it's a darwin->darwin build, I'd expect those to be the darwin packages too.

Reply:
So thinking about this a lot more. This is essentially the fundamental issue that we (hydra) want to evaluate for some platform on some other platform? Or to turn it on it's head, we require arch-independent eval.